Gems/ONNX/Code/Source/Clients/ONNXSystemComponent.cpp

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include "ONNXSystemComponent.h"
+#include <ONNX/Model.h>
+
+#include <AzCore/Serialization/EditContext.h>
+#include <AzCore/Serialization/EditContextConstants.inl>
+#include <AzCore/Serialization/SerializeContext.h>
+
+namespace ONNX
+{
+    void ONNXSystemComponent::AddTimingSample(const char* modelName, float inferenceTimeInMilliseconds, AZ::Color modelColor)
+    {
+        m_timingStats.PushHistogramValue(modelName, inferenceTimeInMilliseconds, modelColor);
+    }
+
+    void ONNXSystemComponent::OnImGuiUpdate()
+    {
+        if (!m_timingStats.m_show)
+        {
+            return;
+        }
+
+        if (ImGui::Begin("ONNX"))
+        {
+            m_timingStats.OnImGuiUpdate();
+        }
+    }
+
+    void ONNXSystemComponent::OnImGuiMainMenuUpdate()
+    {
+        if (ImGui::BeginMenu("ONNX"))
+        {
+            ImGui::MenuItem(m_timingStats.GetName(), "", &m_timingStats.m_show);
+            ImGui::EndMenu();
+        }
+    }
+
+    void ONNXSystemComponent::Reflect(AZ::ReflectContext* context)
+    {
+        if (AZ::SerializeContext* serialize = azrtti_cast<AZ::SerializeContext*>(context))
+        {
+            serialize->Class<ONNXSystemComponent, AZ::Component>()->Version(0);
+
+            if (AZ::EditContext* ec = serialize->GetEditContext())
+            {
+                ec->Class<ONNXSystemComponent>("ONNX", "[Description of functionality provided by this System Component]")
```

Comment:
Should replace the boilderplate description text with something like "Provides ONNX runtime features in O3DE"

Reply:
Changed